hazelcast/src/test/java/com/hazelcast/map/MapTransactionTest.java

```diff
@@ -1329,4 +1327,61 @@ public Object execute(TransactionalTaskContext context) throws TransactionExcept
             }
         });
     }
+
+
+    @Test
+    public void testGetForUpdateLoadsKeyFromMapLoader() {
```

Comment:
Personally I would go over the different permutations.

1: get for updated when locked
2: get for update when item doesn't exist in db.
3: get for update when item does exist in db.

Perhaps there are more combination.

Also use the following naming:

getForUpdate_whenLockedByOtherThread

or with a test prefix.

Going over all permutations helps to discover current + future bugs.